polygeist/tools/cgeist/Test/Verification/sycl/issue7835.cpp

```diff
@@ -0,0 +1,60 @@
+// RUN: clang++ -fsycl -fsycl-device-only -O0 -w -emit-mlir %s -o - | FileCheck %s --check-prefix=CHECK-MLIR
+// RUN: clang++ -fsycl -fsycl-device-only -O3 -w -S -emit-llvm -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o - | FileCheck %s --check-prefix=CHECK-LLVM
+
+#include <sycl/sycl.hpp>
+
+// CHECK-MLIR: gpu.module @device_functions
```

Comment:
Do we care about the MLIR/LLVMIR generated? If not, can this test case be simplified to just ensure it can be compiled successfully?

Reply:
I think having the LLVM IR is better because it checks that the code the compiler generates is what is expected.